Sources/CardPayments/CardClient.swift

```diff
@@ -127,11 +131,29 @@ public class CardClient: NSObject {
                     }
                 }
                 
-                let cardResult = CardResult(orderID: orderId, deepLinkURL: url)
-                self.notifySuccess(for: cardResult)
+                guard let url else {
+                    self.notifyFailure(with: CardClientError.missingDeeplinkURLError)
+                    return
+                }
+                
+                if self.getQueryStringParameter(url: url.absoluteString, param: "state") != nil
+                && self.getQueryStringParameter(url: url.absoluteString, param: "code") != nil {
```

Comment:
It looks like we are doing several additional checks throughout this flow that we are not doing in the [Android PR](https://github.com/paypal/paypal-android/pull/227/files). Is there a reason these checks are different across platforms?

Reply:
I will sync with Steven and Rich today about this.

Reply:
So our reasoning for adding this additional check is in the case that the url is not properly formatted for error or success, we don't want to return success. I added a comment on the Android PR.

Reply:
Thanks for raising that on the Android PR! More just wanted to make sure we were using the same logic across the platforms and it sounds like what exists on iOS in this PR _should_ exist on Android as well.

Reply:
It's a pleasure. Thanks for the feedback. Android has the same logic as ours.